tests/MongoDB.Driver.Tests/Specifications/auth/AuthTestRunner.cs

```diff
@@ -75,9 +79,9 @@ private void AssertValid(MongoCredential mongoCredential, BsonDocument definitio
                 mongoCredential.Source.Should().Be(ValueToString(expectedCredential["source"]));
                 mongoCredential.Mechanism.Should().Be(ValueToString(expectedCredential["mechanism"]));
 
-                var authenticator = mongoCredential.ToAuthenticator();
-                if (authenticator is GssapiAuthenticator gssapiAuthenticator)
+                if (mongoCredential.Mechanism == GssapiAuthenticator.MechanismName)
                 {
+                    var gssapiAuthenticator = mongoCredential.ToAuthenticator() as GssapiAuthenticator;
```

Comment:
Use this instead:

    var gssapiAuthenticator = (GssapiAuthenticator)mongoCredential.ToAuthenticator();

Reply:
Done.